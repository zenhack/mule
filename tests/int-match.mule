#XFAIL: match not implemented
match 1 with
  | 1 -> 1
  | _ -> 0
end
