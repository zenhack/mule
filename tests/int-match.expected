
Desugared:

	((match/const (((Integer 1) (Integer 1))) (_ (fn $6 (Integer 0))))
	 (Integer 1))

inferred type:

	int

Runtime term:

	(App
	 ((ConstMatch (cm_cases (((Integer 1) (Const (Integer 1)))))
	   (cm_default (Lam (0 () (Const (Integer 0))))))
	  (Const (Integer 1))))

Evaluated:

	(Const (Integer 1))
